### PR TITLE
Add helper function to get debug type name

### PR DESCRIPTION
**Pull request by jay-zhuang**

### CODE REVIEW

File: include/rocksdb/utilities/debug.h [hunk 1]
```diff
@@ -25,9 +25,8 @@ struct KeyVersion {
   std::string user_key;
   std::string value;
   SequenceNumber sequence;
-  // TODO(ajkr): we should provide a helper function that converts the int to a
-  // string describing the type for easier debugging.
   int type;
+  std::string GetTypeName();
```
Comment (mrambacher) [inline include/rocksdb/utilities/debug.h:29]: Nit: Declare the function as const

**Review by mrambacher — APPROVED** [review]
LGTM.  Thanks!

File: utilities/debug.cc [hunk 2]
```diff
@@ -8,9 +8,49 @@
 #include "rocksdb/utilities/debug.h"
 
 #include "db/db_impl/db_impl.h"
+#include "rocksdb/utilities/options_type.h"
 
 namespace ROCKSDB_NAMESPACE {
 
+static std::unordered_map<std::string, ValueType> value_type_string_map = {
```
Comment (riversand963) [inline utilities/debug.cc:15]: I wonder if we can put assert/static_assert to make sure this map covers all known types.

**Review by riversand963 — APPROVED** [review]

File: utilities/debug.cc (hunk 2 above)
Comment (jay-zhuang) [reply utilities/debug.cc:15]: I added a simple unittest to make sure the list is up to date.

**Pull request closed without merging**